Add statistical card to extensibility (#2962)
kyma-project/busola: docs/extensibility/50-list-and-details-widgets.md

### `ControlledBy`

The `ControlledBy` widgets render the kind and the name with a link to the resources that the current resource is

> dependent on.

Reviewer: There should not be new line in here

> ### ConditionList

Reviewer 2: ```suggestion
### `ConditionList`
```

> The condition List widget renders the conditions as an expandable list with condition details. This widget is primarily designed to be used in overview section **data.details.status**

Reviewer 2: ```suggestion
The `ConditionList` widget renders the conditions as an expandable list with condition details. This widget is primarily designed for the overview section **data.details.status**
```

### `FeaturedCard`

> The `FeaturedCard` widgets render a promotional banner with its own title, description, illustration, and a maximum of 2

Reviewer: Something is wrong with formating. There should not be new line in here

additional children.
The `FeaturedCard` can be closed in the top right corner.
It is important to select a unique ID for each `FeaturedCard`, since it will be displayed/hidden based on its ID.

**TIP:** It is **recommended** to use the `FeatureCard` widget via [injections](./70-widget-injection.md) (**slot:

> banner**). This will turn all `FeatureCards` at the top of a section into a carousel-component. Nevertheless, it is also

Reviewer: Why new line?

> possible to use it as a regular widget.

Reviewer: Why new line?

### `Plain`

The `Plain` widgets render all contents of an object or list sequentially without any decorations. This is the default

> behavior for all objects and arrays.

Reviewer: Why new line?

### `ResourceList`

The `ResourceList` widgets render a list of Kubernetes resources. The ResourceList widgets should be used along with

> the [related resources](datasources-section.md).

Reviewer: Why new line?

| **search.source**         | **Yes**  | [JSONata](jsonata.md) expression       | Used to fetch data for the column. In its simplest form, it's the path to the value.                                                                                                                                                                       |
| **search.searchFunction** | No       | [JSONata](jsonata.md) search function  | It allows you to use the `$input` variable to get the search input's value that can be used to search for more complex data.                                                                                                                               |

Since the `ResourceList` widget does more than just list the items, you must provide the whole data

> source (`$myResource()`) instead of just the items (`$myResource().items`).

Reviewer: Why new line?

### `ResourceRefs`

The `ResourceRefs` widgets render the lists of links to the associated resources. The corresponding specification object

> must be an array of objects `{name: 'foo', namespace: 'bar'}`.

Reviewer: Why new line?

<img src="./assets/display-widgets/ResourceRefs.png" alt="Example of a ResourceRefs widget" style="border: 1px solid #D2D5D9">

> ### StatisticalCard

Reviewer 2: ```suggestion
### `StatisticalCard`
```

StatisticalCard widgets render a card component with a several numerical information elements.

> This widget is primarily designed to be used in Monitoring and Health section **data.details.health** or via [injections](#widget-injections-overview) (**destination: ClusterOverview, slot: health**), allowing the card to be rendered within the dense grid layout of the ClusterOverview's Monitoring and Health section.

Reviewer 2: ```suggestion
To display the widget in the **Monitoring and Health** section of a details page, configure it in **data.details.health**.  
To render the card within the dense grid layout in the **Monitoring and Health** section of **Cluster Details**, use [injections](#widget-injections-overview) (**destination: ClusterOverview, slot: health**).
```

Reviewer 2: ![Screenshot 2024-06-21 at 11 56 59](https://github.com/kyma-project/busola/assets/23300533/08ea4b83-5124-4b76-aea0-a3bd750ae2c5)
Is it Cluster Overview or Cluster Details? PTAL at the screenshot.

Reviewer: In navigation it's Cluster Details

These are the available `StatisticalCard` widget parameters:

| Parameter            | Required | Type                                       | Description                                                                                                                                                                                                          |
| -------------------- | -------- | ------------------------------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | **mainValue**        | **Yes**  | object                                     | Main value displayed with bigger font                                                                                                                                                                                |

Reviewer 2: ```suggestion
| **mainValue**        | **Yes**  | object                                     | The main value displayed using a bigger font.                                                                                                                                                                              |
```

> | **mainValue.source** | **Yes**  | string or [JSONata](jsonata.md) expression | It is used to fetch data for the column. In its simplest form, it's the path to the value.                                                                                                                           |

Reviewer 2: ```suggestion
| **mainValue.source** | **Yes**  | string or [JSONata](jsonata.md) expression | Fetches data for the column. In its simplest form, it's the path to the value.                                                                                                                           |
```

> | **mainValue.name**   | **Yes**  | string                                     | Name for the primary label of this field. Required for most widgets (except for some rare cases that don't display a label). This can be a key to use from the [**translation** section](./translations-section.md). |

Reviewer 2: ```suggestion
| **mainValue.name**   | **Yes**  | string                                     | The name for the primary label of this field. Required for most widgets (except for some rare cases that don't display a label). This can be a key to use from the [**translation** section](./translations-section.md). |
```

> | **children**         | No       | array of objects                           | Array of additional values, listed next to the main one                                                                                                                                                              |

Reviewer 2: ```suggestion
| **children**         | No       | array of objects                           | An array of additional values, listed next to the main one.                                                                                                                                                              |
```

> | **children.source**  | **Yes**  | string or [JSONata](jsonata.md) expression | It is used to fetch data for the column. In its simplest form, it's the path to the value.                                                                                                                           |

Reviewer 2: ```suggestion
| **children.source**  | **Yes**  | string or [JSONata](jsonata.md) expression | Fetches data for the column. In its simplest form, it's the path to the value.                                                                                                                           |
```

> | **children.name**    | **Yes**  | string                                     | Name for the primary label of this field. Required for most widgets (except for some rare cases that don't display a label). This can be a key to use from the [**translation** section](./translations-section.md). |

Reviewer 2: ```suggestion
| **children.name**    | **Yes**  | string                                     | The name for the primary label of this field. Required for most widgets (except for some rare cases that don't display a label). This can be a key to use from the [**translation** section](./translations-section.md). |
```

> |

Reviewer 2: ```suggestion

```

> This is an example for widget usage in the **data.details.health** section which allows StatisticalCard to be displayed in the details page in the Monitoring and Health section:

Reviewer 2: ```suggestion
This is an example of the widget configuration in the **data.details.health** section which allows the `StatisticalCard` to be displayed on the details page in the **Monitoring and Health** section:
```

```yaml
details: |-
  health:
    - name: MyTitle
      widget: StatisticalCard
      source: status
      mainValue:
        name: MySubtitle
        source: $item.importantValue
      children:
        - name: ExtraInformation1
          source: $item.value1
        - name: ExtraInformation2
          source: $item.value2
```

> This is an example for widget usage via injection which allows StatisticalCard to be displayed in ClusterOverview's Monitoring and Health section:

Reviewer 2: ```suggestion
This is an example of the widget configured using injection which allows the `StatisticalCard` to be displayed in the **Monitoring and Health** section of **Cluster Details**:
```

### `Table`

Table widgets display array data as rows of a table instead of free-standing components. The **children** parameter
defines the values used to render the columns. Similar to the **list** section of the ConfigMap, you should use inline

> widgets only as children.

Reviewer: Why new lines?

```

<img src="./assets/display-widgets/Tabs.png" alt="Example of a tabs widget" style="border: 1px solid #D2D5D9">

### StatisticalCard

StatisticalCard widgets render a card component with a several numerical information elements.

> This widget is primarily designed to be used via [injections](#widget-injections-overview) (**destination: ClusterStats,

Reviewer: It can be used in data.details.status as well. Would be good to docs for it too

slot: cards**), allowing the card to be rendered within the dense grid layout of the ClusterOverview's statistical cards
section.

These are the available `StatisticalCard` widget parameters:

| Parameter             | Required | Type    | Description                                                                  |
| --------------------- | -------- | ------- | ---------------------------------------------------------------------------- |
| **resourceURL**       | **No**   | string  | Path to busola resource.                                                     |
| **isClusterResource** | **No**   | boolean | Defines if resource is cluster wide                                          |
| **allNamespaceURL**   | **No**   | boolean | Defines if resource URL should point to a list of resource in all namespaces |

#### Example

```yaml

> injections: |-

Reviewer: Second example for use in data.details.status
